tests/4.5/taskloop/test_taskloop_shared.c

```diff
@@ -0,0 +1,41 @@
+//===--test_taskloop_shared.c ----------------------------------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// Test uses a value within a taskloop & an atomic construct to update the value of
+// s_val. The shared clause ensures that the s_val will be shared between threads,
+// and therefore should be equal to N After the taskloop region.
+//
+////===--------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int taskloop_shared() {
+
+  int errors = 0;
+  int s_val=0;
+  int check = -1;
+
+  #pragma omp target map(tofrom: check, s_val)//map(tofrom: s_val) map(from: check)
+  {
+    check = 7;
+    #pragma omp taskloop shared(s_val)
+      for (int i = 0; i < N; ++i){
+        ++s_val; 
```

Comment:
This looks unsafe. Don't you need something like 'omp atomic update' — or 'omp taskloop reduction(+:s_val)' here?